loader/loader.c

```diff
@@ -3534,13 +3540,32 @@ VkResult loader_icd_scan(const struct loader_instance *inst, struct loader_icd_t
                                "loader_icd_scan: ICD JSON %s does not have an \'api_version\' field.", file_str);
                 }
                 VkResult icd_add_res = VK_SUCCESS;
-                icd_add_res = loader_scanned_icd_add(inst, icd_tramp_list, fullpath, vers);
+                enum loader_layer_library_status lib_status;
+                icd_add_res = loader_scanned_icd_add(inst, icd_tramp_list, fullpath, vers, &lib_status);
                 if (VK_ERROR_OUT_OF_HOST_MEMORY == icd_add_res) {
                     res = icd_add_res;
                     goto out;
                 } else if (VK_SUCCESS != icd_add_res) {
-                    loader_log(inst, VULKAN_LOADER_ERROR_BIT | VULKAN_LOADER_DRIVER_BIT, 0,
-                               "loader_icd_scan: Failed to add ICD JSON %s.  Skipping ICD JSON.", fullpath);
+                    switch (lib_status) {
+                        case LOADER_LAYER_LIB_NOT_LOADED:
```

Comment:
It's not really a fall-through, is it?  It's part of the same block.  Fall through is when you do something under a block and then fall through to the next.

Reply:
Hmm in my mind I don't differentiate empty and non-empty case statements, so this would be a fall through in my book. However, the C++ Core Guidelines are fine with this situation so I think this code is fine to leave as is. Guidelines in question.
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es78-dont-rely-on-implicit-fallthrough-in-switch-statements